Changes in support of Porous Weakly Compressible Finite Volume

GiudGiud opened the pull request:
- allow for density variable for conserved variable sets
- uses NS namespace more in PINSFV
- allow to use moosefunctors more often over matprop / variables
- rework energy diffusion kernels around anisotropy rather than diffusivity / effective diffusivity
- rename some kernels to make distinction between old matprops and new ones used by CNSFV and INSFV respectively

GiudGiud committed:
Fix effective diffusion to still have an epsilon term in diffusivity

GiudGiud committed:
Add flexibility around density in functorfluidprops, as some variable sets may already define it, refs #16809

GiudGiud committed:
Add functor matprop energy time derivative kernel for PINSFV, refs #16756

GiudGiud committed:
Re-order energy diffusion kernel around anisotropy rather than whether effective or regular diffusivity, refs #17088

GiudGiud committed:
Use NS namespace more in PINSFV
Fixup functor properties when density is a variable

GiudGiud committed:
Add 2D porous weakly compressible test, refs #16756 #18806
Clang syntax
Add NS.h header for unity build

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/857374/) on 7399ba4 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/19170/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/19170/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format aca4e1e9afab806da7f13979859fbb282b41b5fe`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/857424/) on 60cb497 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/19170/site)

This comment will be updated on new commits.

GiudGiud committed:
Add documentation for new split kernels
Adapt documentation for effective/regular diffusion to isotropic/anisotropic and add a little more than look at pronghorn manual

GiudGiud committed:
Add FV citing papers for NS

lindsayad reviewed (changes requested):
I really like this PR for a couple reasons
- Continuing to bifurcate INSFV and CNSFV in terms of the residual objects. Apart from how they currently differ with their use of regular/functor properties (which is a transient thing), they will *have* to differ in the future with the Rhie-Chow interpolator, so I think it's great that we're continuing to develop the distinct objects
- Though we are bifurcating the residual objects, we are adding more consistency on parameter naming. Love it

GiudGiud committed:
Apply suggestions from code review
Clang format of patch

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

lindsayad committed:
Protect against invalid functor params with isParamValid

Previously we just checked whether the parameters had the parameter, not
whether it had been set. So now we make sure we check that

Closes #19171

GiudGiud committed:
Add missing doc file

GiudGiud committed:
Remove old material property system kernels that were moved to CNSFV. CNSFV is happy enough without them, they kind of dont fit very well with these Euler solvers, being diffusion kernels

GiudGiud committed:
Add test for solving porous flow NS equations with pressure integrated by parts, closes #18478

Protect against invalid mat-prop name params with isParamValid

Use adCoupledValue instead of coupledValue

fixup! Use adCoupledValue instead of coupledValue

lindsayad reviewed (changes requested):
I thought about this while I was falling asleep… in places you call adGradSln, you should catch the return as const auto &. This will prevent copying with our current caching system and will still work even if/when we move off the caching system and return a new object.

lindsayad reviewed (changes requested):
Guess it was only one place!

GiudGiud committed:
Alex review:
Use references to avoid a copy of the AD gradients
Dont admit to failure at r=0 when it actually works now that I dont forget the Jacobian like a dummy

GiudGiud committed:
Rename p/r term back to PNSFV MomentumPressureFluxRZ and transition it to functors
PCNSFV needs this kernel, with matprops
PINSFV needs this kernel, with variables or functors, but not matprops
Only functors can do both
PINSFV will need its own version in the future for the RC UO. This doesnt have to happen now

lindsayad reviewed (approved):
I was torn on the `PNSFVMomentumPressureFluxRZ` naming, but when we introduce the RC user object we will have to re-write that input block anyways (`momentum_component` will be a required parameter for all (P)INSFV momentum residual objects), so I don't think there's much additional damage in changing `type` in the process.
